scripts/deployment-test/cf-pages.mjs

```diff
@@ -86,8 +88,46 @@ async function createCloudflareDeployment() {
     if (promise.headers.get("Content-Type").includes("application/json")) {
       console.error(await promise.json());
     }
-    throw new Error(`Failed to create cloudflare pages project`);
+    throw new Error(`Failed to create Cloudflare Pages project`);
   }
+
+  let deployment = await promise.json();
+
+  return deployment.result.id;
+}
+
+function checkDeploymentStatus() {
+  let operation = retry.operation({ retries: 20 });
+  let url = `https://api.cloudflare.com/client/v4/accounts/${process.env.CF_ACCOUNT_ID}/pages/projects/${APP_NAME}/deployments`;
+
+  return new Promise((resolve, reject) => {
+    operation.attempt(async (currentAttempt) => {
+      console.log(`Checking deployment status; attempt ${currentAttempt}`);
+      let response = await fetch(url, {
+        headers: {
+          "X-Auth-Email": process.env.CF_EMAIL,
+          "X-Auth-Key": process.env.CF_GLOBAL_API_KEY,
+        },
+      });
+
+      if (response.status >= 200 && response.status < 400) {
+        let data = await response.json();
+        let deployment = data.result[0];
```

Comment:
each run creates a new GitHub repo and Pages project, so this is _fine_ to do, we could also return the deployment id from `createCloudflareDeployment` and pass it to this function